api/src/main/java/org/pmiops/workbench/api/CohortReviewController.java

```diff
@@ -800,29 +801,10 @@ private DbCohortReview createNewCohortReview(
         .creator(creator);
   }
 
-  /**
-   * Helper method that will populate all gender, race and ethnicity per the spcecified list of
-   * {@link DbParticipantCohortStatus}.
-   *
-   * @param participantCohortStatuses
-   */
-  private void lookupGenderRaceEthnicityValues(
-      List<DbParticipantCohortStatus> participantCohortStatuses) {
-    List<DbCriteria> criteriaList = cbCriteriaDao.findGenderRaceEthnicity();
-    participantCohortStatuses.forEach(
-        pcs -> {
-          pcs.setRace(getName(criteriaList, pcs.getRaceConceptId()));
-          pcs.setGender(getName(criteriaList, pcs.getGenderConceptId()));
-          pcs.setEthnicity(getName(criteriaList, pcs.getEthnicityConceptId()));
-        });
-  }
-
-  private String getName(List<DbCriteria> criteriaList, Long conceptId) {
-    List<DbCriteria> returnList =
-        criteriaList.stream()
-            .filter(c -> c.getConceptId().equals(conceptId.toString()))
-            .collect(Collectors.toList());
-    return returnList.isEmpty() ? null : returnList.get(0).getName();
+  /** Build a map that contains all gender/race/ethnicity names with the concept id as the key. */
+  private Map<Long, String> getGenderRaceEthnicityMap() {
```

Comment:
So could you still have a transient method to get the gender directly(that uses this map)? I.e. if you don't know or care about the conceptId?

Reply:
I suppose we could, but not sure how easy this would be to implement with a Hibernate Entity since i each entity is back by 1 table. This gender/race/eth map is backed by a different table/sql statement and is part of the cdr and can change per cdr version.